style-packages.md

```diff
@@ -0,0 +1,255 @@
+according to the HTTP aspect the file handles.
+In the following example, an HTTP package is broken down into a few files:
+header types and code, cookie types and code, the actual HTTP implementation, and
+documentation of the package.
+
+```
+- doc.go       // package documentation
+- headers.go   // HTTP headers types and code
+- cookies.go   // HTTP cookies types and code
+- http.go      // HTTP client implementation, request and response types, etc.
+```
+
+### Keep types close
+
+As a rule of thumb, keep types closer to where they are used. This makes it easy for
+any maintainer (not just the original author) to find a type.
+A good place for a Header struct type might be in `headers.go`.
+
+```
+$ cat headers.go
+
+// Header represents an HTTP header.
+type Header struct {...}
+```
+
+### Use cmd directory for commands
+
+If your repository is large and provides multiple commands via multiple main packages,
+put the these main packages under a `cmd` directory. Go users are familiar with the convention
+that a cmd directory will provide main packages.
+
+For example, [x/tools](https://godoc.org/golang.org/x/tools/)
+publishes all of the tools from [golang.org/x/tools/cmd](https://godoc.org/golang.org/x/tools/cmd).
+
+### Optimize for godoc
+
+It is a great exercise to use godoc in the early phases of your package's API design to see
+how your concepts will be rendered on doc. Sometimes, the visualization also has an impact
+on the design. Godoc is the way your users will consume a package, so it is ok to tweak
+things to make them more accessible. Run `godoc -http=<hostport>` to start a godoc server locally.
+
+### Provide examples to fill the gaps
+
+In some cases, you may not be able to provide all related types from a single package. It might be noisy
+to do so, or you might want to publish concrete implementations of a common interface from a separate
+package, or those types could be owned by a third-party package.
+Give examples to help the user to discover and understand how they are used together.
+
+```
+$ godoc cloud.google.com/go/datastore
+func NewClient(ctx context.Context, projectID string, opts ...option.ClientOption) (*Client, error)
+...
+```
+
+NewClient works with option.ClientOptions but it is neither the datastore package
+nor the option package that export all the option types.
+
+```
+$ godoc google.golang.org/extraoption
+func WithCustomValue(v string) option.ClientOption
+...
+```
+
+If your API requires many non-standard packages to be imported, it is often useful to add
+a [Go example](https://blog.golang.org/examples) to give your users some working code.
+
+Examples are a good way to increase visibility of a less discoverable package.
+For example, an example for datastore.NewClient might reference the extraoption package.
+
+### Don't export from main
+
+Don't export symbols from a main package if you are not building the package
```

Comment:
Many Go users would be confused by this as they are unfamiliar with what "export symbols" means and wouldn't know if they are doing it or not. Add an introduction with context for when they may do this would help a lot.